templates/addons/etcd-backup-restore/etcd-backup-restore.yaml

```diff
@@ -0,0 +1,191 @@
+          app.kubernetes.io/name: etcd
+      updateStrategy:
+        type: RollingUpdate
+      template:
+        metadata:
+          labels:
+            app.kubernetes.io/name: etcd
+        spec:
+          initContainers:
+          - name: create-creds-file
+            image: busybox
+            command:
+            - ash
+            - -c
+            - |
+              cat << EOF > /data/${CREDS_FILENAME}
+              {
+                "accessKeyID": "$${AWS_ACCESS_KEY_ID}",
+                "secretAccessKey": "$${AWS_SECRET_ACCESS_KEY}",
+                "region": "$${AWS_REGION}",
+                "endpoint": "$${AWS_REGION}.linodeobjects.com",
+                "sseCustomerKey": "$${AWS_SSE_CUSTOMER_KEY}"
+              }
+              EOF
+            volumeMounts:
+              - name: data-volume
+                mountPath: /data
+            env:
+            - name: "AWS_REGION"
+              value: ${OBJ_BUCKET_REGION}
+            - name: "AWS_ACCESS_KEY_ID"
+              valueFrom:
+                secretKeyRef:
+                  name: etcd-backup
+                  key: "access_key_rw"
+            - name: "AWS_SECRET_ACCESS_KEY"
+              valueFrom:
+                secretKeyRef:
+                  name: etcd-backup
+                  key: "secret_key_rw"
+            - name: "AWS_SSE_CUSTOMER_KEY"
+              valueFrom:
+                secretKeyRef:
+                  name: sse-key
+                  key: sse-key
+          containers:
+          - name: backup-restore
+            command:
+            - /etcdbrctl
+            - snapshot
+            - --schedule=0 0 * * *
+            - --garbage-collection-period=1m
+            - --storage-provider=S3
+            - --store-prefix=etcd-backup
+            - --insecure-transport=true
+            - --insecure-skip-tls-verify=true
+            - --endpoints=https://127.0.0.1:2379
+            - --defragmentation-schedule=0 0 */3 * *
+            - --etcd-connection-timeout=30s
+            - --delta-snapshot-period=60s
+            - --delta-snapshot-memory-limit=104857600
+            - --compress-snapshots=true        
+            - --compression-policy=gzip
+            - --cacert=/etc/kubernetes/pki/etcd/ca.crt
+            - --cert=/etc/kubernetes/pki/etcd/healthcheck-client.crt
+            - --key=/etc/kubernetes/pki/etcd/healthcheck-client.key
+            image: ${ETCDBR_IMAGE}
+            securityContext:
+              allowPrivilegeEscalation: false
+              runAsUser: 0
+            imagePullPolicy: Always
+            ports:
+            - containerPort: 8080
+              name: server
+              protocol: TCP
+            resources:
+              limits:
+                cpu: 100m
+                memory: 1Gi
+              requests:
+                cpu: 23m
+                memory: 128Mi
+            env:
+            - name: AWS_APPLICATION_CREDENTIALS_JSON
+              value: ${CREDS_DIRNAME}${CREDS_FILENAME}
```

Comment:
Can this just be hard coded? any reason a user would want to configure this?

Reply:
It certainly can be. The directory name is being used for the mountPath so I figured we could split it but that just increases the number of env vars we need so I'll hardcode it

Reply:
Done. Hard-coded